mesonbuild/_pathlib.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import typing as T
+
+# Python 3.5 does not have the strict kwarg for resolve and always
+# behaves like calling resolve with strict=True in Python 3.6+
+#
+# This module emulates the behavior of Python 3.6+ by in Python 3.5 by
+# overriding the resolve method with a bit of custom logic
+#
+# TODO: Drop this module as soon as Python 3.5 support is dropped
+
+if T.TYPE_CHECKING:
+    from pathlib import Path
+else:
+    if sys.version_info.major <= 3 and sys.version_info.minor <= 5:
```

Comment:
I think this can just be  `if sys.version_info <= (3, 5)`

Reply:
At least not in this form. Comparing with a tuple is apparently not that straightforward...

```
Python 3.5.10 (default, Sep  6 2020, 09:43:22) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version_info <= (3, 5)
False
>>> sys.version_info <= (3, 4)
False
>>> sys.version_info <= (3, 6)
True
>>> sys.version_info <= (3, 5, 10)
False
>>> sys.version_info <= (3, 5, 11)
True
```

So we would have to compare to `(3, 5, 999)` which is worse than the explicit `major` and `minor` solution IMHO because of the magic number. I am not 100% sure if `< (3, 6)` will always work, so I would like to keep the current version since it has the least potential for surprises.

Reply:
Okay